include/deal.II/matrix_free/cuda_matrix_free.templates.h

```diff
@@ -103,6 +105,10 @@ namespace CUDAWrappers
       cudaError_t error_code = cudaMalloc(array_device, n * sizeof(Number1));
       AssertCuda(error_code);
 
+      // TODO This is very dangerous because we are doing a memcopy but with
+      // different data types. However this is very useful to move Point to the
+      // device where are they are store as Tensor. Thus, we need Point on the
+      // device in order to make this function safer.
```

Comment:
I guess this comment would be better where this function is actually used.

Some typos:
However, this is very useful to move Point to the device
where they are stored as Tensor. Thus, we need Point 
on the device in order to make this function safer.